source/common/http/alternate_protocols_cache_impl.cc

```diff
@@ -28,15 +21,25 @@ AlternateProtocolsCacheImpl::findAlternatives(const Origin& origin) {
         nullptr);
   }
 
-  const Entry& entry = entry_it->second;
-  if (time_source_.monotonicTime() > entry.expiration_) {
-    // Expire the entry.
-    // TODO(RyanTheOptimist): expire entries based on a timer.
+  std::vector<AlternateProtocol>& protocols = entry_it->second;
+
+  const MonotonicTime now = time_source_.monotonicTime();
+  auto it = protocols.begin();
```

Comment:
should we add bounds checks on the number of returned tokens we accept per returned header?
I think Envoy default header limits are 64k and that allows for an unreasonable number of tokens per lookup.

in this PR it might be all we do is have a defined constant (say 10?) and assert we don't insert more, and we'd do the validation in your e2e PR and drop overly large entries.